Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1166,14 +1238,56 @@ void MediaPlayerPrivateGStreamer::notifyPlayerOfAudio()
         g_signal_emit_by_name(m_pipeline.get(), "get-audio-pad", i, &pad.outPtr(), nullptr);
         ASSERT(pad);
 
-        String streamId = "A" + String::number(i);
+        auto probeData = adoptRef(*new StreamStartProbeData(this, static_cast<int>(i)));
+        probeData->lock.lock();
```

Comment:
I explicitly lock before adding the probe, otherwise it might be called with stream-start, fail to find the track object and bail out, meaning the ID might never get updated.